### ISSUE TITLE
Sumatra shows wrong names of embedded files

### ISSUE THREAD

**Issue by u-fischer** [sumatrapdfreader/sumatrapdf#1666]
Embedded files that should appear in the "attachment panel" should have their /FileSpec dictionary  referenced in a name tree in the pdf. The keys in this name tree can be freely choosen (but should be sorted lexically). The problem is that sumatra uses this keys when showing the attachements instead of using the /F or /UF entry in the /FileSpec dictionary:

![image](https://user-images.githubusercontent.com/4047173/87058340-40bee280-c208-11ea-99cb-6059869795ab.png)

In the adobe reader the embedded files are shown like this:

![image](https://user-images.githubusercontent.com/4047173/87059578-c1321300-c209-11ea-8d5a-8ba00837029d.png)

[embedfiles.pdf](https://github.com/sumatrapdfreader/sumatrapdf/files/4897824/embedfiles.pdf)

**Comment by GitHubRulesOK** [comment]
Ulrike
There have been many issues over embedment recently addressed since version 3.2 the current pre-release shows them as follows
![image](https://user-images.githubusercontent.com/12754913/87066327-47e6f000-c20a-11ea-8df9-cc141a40c683.png)

The files can be saved as shown (they will currently not be opened by SumatraPDF) the sort order is as currently in the order of objects found e.g. 6 = blub & 8 = abc

@kjk 
I think the big/small sort order is unclear as to intent ?

**Comment by u-fischer** [comment]
Oh, that looks much better ;-). Nice that this has been already addressed. 

Regarding the sort order: I don't think that the reference say something about it, but imho the most logical is to follow the order in the name tree - it is easier to control this then the actual object number (which are sometimes reserved before use).

**Comment by GitHubRulesOK** [comment]
Ulrike
I think acrobat allows for alpha sorting, but then this is not the same size team as adobe.
There are still issues arround file extraction or auto running pdf embedments due to the masive internal MuPDF changes.

PS the excessive white space between glyphs is also recently improved in pre-release as the issue was mainly related to upstream freetype issues

**Comment by u-fischer** [comment]
Yes, adobe allows sorting. But this is not really needed. Imho it would be quite okay, if sumatra simply follows the order in the name tree.

**Comment by GitHubRulesOK** [comment]
@kjk there has been some regression as this was generally working in 3.5.2 but not in 3.6 or later

<img width="789" height="323" alt="Image" src="https://github.com/user-attachments/assets/b8aa6459-7b44-4235-a2cb-9af42c376100" />

There is no indication that there are attached files thus no way to extract via selection. This may also impact embedded PDFs ?
<img width="877" height="154" alt="Image" src="https://github.com/user-attachments/assets/c4dbc7ce-faae-4105-99e8-d1ed5f405b51" />
